Release 2.1.2 rc1

osalyk opened the pull request:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/daos-stack/pmdk/19)
<!-- Reviewable:end -->

jolivier23 commented on packaging/Dockerfile.mockbuild:
```diff
@@ -1,5 +1,5 @@
 #
-# Copyright 2018-2019, Intel Corporation
+# Copyright 2018-2020, Intel Corporation
```
2020?

jolivier23 replied:
so are you planning to update this?

JohnMalmberg commented:
Obsolete

osalyk committed:
test: call setup before require_sds

It must be called after setup because it relies
on the contents of PMEMOBJ_CONF which is set there.

osalyk committed:
test: unblock obj_sds tests for non-pmem

osalyk committed:
test: check if mock functions are used

janekmi committed:
Merge pull request #6152 from osalyk/setup_sds

test: call setup before require_sds

janekmi committed:
common: update before install (aarch64)

Signed-off-by: Jan Michalski <jan-marian.michalski@hpe.com>

osalyk committed:
Merge pull request #6155 from pmem/main-aarch64-fix

common: update before install (aarch64)

janekmi committed:
Merge pull request #6153 from osalyk/sds_non-pmem

test: unblock obj_sds tests for non-pmem

janekmi committed:
Merge pull request #6154 from osalyk/obj_sds-test

test: check if mock functions are used

osalyk committed:
common: update workflow files

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

janekmi committed:
Merge pull request #2 from daos-stack/osalyk/workflow_files

common: update workflow files

osalyk committed:
test: add unit test for sds_check (#1)

+ minor adjustments to error/warning messages

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

janekmi committed:
common: expand the sds.at_create CTL to also cover pmemobj_open()

https://daosio.atlassian.net/browse/DAOS-17449

Ref: daos-stack/pmdk#5

Unclean cherry-pick of 0bf7f8a565aae101f17aec4bf1c168bda4fa1243

Signed-off-by: Jan Michalski <jan-marian.michalski@hpe.com>

osalyk committed:
utils: switch self hosted runners to the new repo location
replace the outdated include command with the import_tasks command

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: switch self hosted runners to the new repo location
replace the outdated include command with the import_tasks command

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
Merge pull request #7 from daos-stack/update_history

common: switch self hosted runners to the new repo location

osalyk committed:
common: fix typo

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
Merge pull request #8 from daos-stack/osalyk/typo

common: fix typo

osalyk committed:
common: split the pmem_tests.yml file (#9)

Due to limited hardware availability, the previous monolithic workflow could not meet the 24-hour limit imposed by GitHub.

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: reshuffle the jobs to fit both workflows into the 24-hour window

Reshuffle the jobs to fit a workflow containing the longest-running jobs into a 24-hour window.

Hence there is no longer a clear cut (with Valgrind vs without Valgrind).
They are simply called part 1 and part 2.

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: extend jobs timeouts (#11)

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: add new timeout to static build

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: update the badges in the README.md file

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
Merge pull request #12 from daos-stack/workflows_4

common: add fixes to workflow files

osalyk committed:
common: limit Static builds to ICX runners only

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

janekmi committed:
test: expand the sds.at_create CTL to also cover pmemobj_open() (fix)

- add missing log entry

Ref: daos-stack/pmdk#14

Signed-off-by: Jan Michalski <jan-marian.michalski@hpe.com>

osalyk committed:
Merge pull request #15 from janekmi/DAOS-17449-expand_sds_at_create-master-fix

test: expand the sds.at_create CTL to also cover pmemobj_open() (fix)

grom72 reviewed (changes requested):
> @grom72 reviewed 1 of 82 files at r1.
> *[Reviewable](https://reviewable.io/reviews/daos-stack/pmdk/19)* status: 1 of 82 files reviewed, 1 unresolved discussion (waiting on @janekmi)

___
*[`ChangeLog` line 12 at r1](https://reviewable.io/reviews/daos-stack/pmdk/19#-Oc66K401c2M9SuCXM68:-Oc66K416lnBmR2dKOeW:b-yf47do) ([raw file](https://github.com/daos-stack/pmdk/blob/1b9ce120abaa174a698e04b17dd5b2b65d28c284/ChangeLog#L12)):*
> ```
> 	  - Conversely, pmemobj_open() does not issue a warning when attempting to open a pool with SDS disabled
> 	  while the feature is force-disabled.
> ```

Worth to mention that
- new home for pmdk software has been set up
- SDS tests has been fixed/improved/etc


<!-- Sent from Reviewable.io -->

grom72 reviewed (changes requested):
> *[Reviewable](https://reviewable.io/reviews/daos-stack/pmdk/19)* status: 1 of 82 files reviewed, 1 unresolved discussion (waiting on @janekmi and @osalyk)

___
*[`ChangeLog` line 12 at r1](https://reviewable.io/reviews/daos-stack/pmdk/19#-Oc66K401c2M9SuCXM68:-Oc66p-LAsPoRm3swMzG:b-4r0a03) ([raw file](https://github.com/daos-stack/pmdk/blob/1b9ce120abaa174a698e04b17dd5b2b65d28c284/ChangeLog#L12)):*
<details><summary><i>Previously, grom72 (Tomasz Gromadzki) wrote…</i></summary><blockquote>

Worth to mention that
- new home for pmdk software has been set up
- SDS tests has been fixed/improved/etc
</blockquote></details>

- PMem HW tests has been reorganized


<!-- Sent from Reviewable.io -->

grom72 reviewed (commented):
> @grom72 reviewed 81 of 82 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/daos-stack/pmdk/19)* status: all files reviewed, 1 unresolved discussion (waiting on @janekmi and @osalyk)



<!-- Sent from Reviewable.io -->

osalyk committed:
common: update Changelog.md (#20)

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: 2.1.2-rc1 release

Signed-off-by: Oksana Salyk <oksana.salyk@hpe.com>

osalyk committed:
common: git versions

osalyk reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/daos-stack/pmdk/19)* status: 81 of 82 files reviewed, 1 unresolved discussion (waiting on @grom72 and @janekmi)

___
*[`ChangeLog` line 12 at r1](https://reviewable.io/reviews/daos-stack/pmdk/19#-Oc66K401c2M9SuCXM68:-Oc_gw5N7OIiB6_wsKDn:b-896fix) ([raw file](https://github.com/daos-stack/pmdk/blob/1b9ce120abaa174a698e04b17dd5b2b65d28c284/ChangeLog#L12)):*
<details><summary><i>Previously, grom72 (Tomasz Gromadzki) wrote…</i></summary><blockquote>

- PMem HW tests has been reorganized
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

janekmi reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> @janekmi reviewed 81 of 82 files at r1, 1 of 1 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/daos-stack/pmdk/19#-:-Oc_oMpC6XrmyHot-2QN:bnfp4nl)* status: all files reviewed, 1 unresolved discussion (waiting on @grom72)



<!-- Sent from Reviewable.io -->

grom72 reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> @grom72 reviewed all commit messages.
> *[Reviewable](https://reviewable.io/reviews/daos-stack/pmdk/19#-:-OcfGfbD0pUEIBRXSaU8:bnfp4nl)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @osalyk)



<!-- Sent from Reviewable.io -->

grom72 reviewed (approved):
> @grom72 reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/daos-stack/pmdk/19)* status: <span>:shipit:</span> complete! all files reviewed, all discussions resolved (waiting on @osalyk)



<!-- Sent from Reviewable.io -->
